d1.go

```diff
@@ -0,0 +1,197 @@
+		res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+		if err != nil {
+			return []D1Database{}, &ResultInfo{}, fmt.Errorf("%s: %w", errMakeRequestError, err)
+		}
+
+		err = json.Unmarshal(res, &r)
+		if err != nil {
+			return []D1Database{}, &ResultInfo{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+		}
+		databases = append(databases, r.Result...)
+		params.ResultInfo = r.ResultInfo.Next()
+		if params.ResultInfo.Done() || !autoPaginate {
+			break
+		}
+	}
+	return databases, &r.ResultInfo, nil
+}
+
+// CreateD1Database creates a new database for an account.
+//
+// API reference: https://developers.cloudflare.com/api/operations/cloudflare-d1-create-database
+func (api *API) CreateD1Database(ctx context.Context, rc *ResourceContainer, params CreateD1Parameters) (D1Database, error) {
+	if rc.Identifier == "" {
+		return D1Database{}, ErrMissingAccountID
+	}
+	uri := fmt.Sprintf("/accounts/%s/d1/database", rc.Identifier)
+	res, err := api.makeRequestContext(ctx, http.MethodPost, uri, params)
+	if err != nil {
+		return D1Database{}, fmt.Errorf("%s: %w", errMakeRequestError, err)
+	}
+
+	var r D1DatabaseResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return D1Database{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return r.Result, nil
+}
+
+// DeleteD1Database deletes a database for an account.
+//
+// API reference: https://developers.cloudflare.com/api/operations/cloudflare-d1-delete-database
+func (api *API) DeleteD1Database(ctx context.Context, rc *ResourceContainer, databaseID string) error {
+	if rc.Identifier == "" {
+		return ErrMissingAccountID
+	}
+	if databaseID == "" {
+		return ErrMissingDatabaseID
+	}
+	uri := fmt.Sprintf("/accounts/%s/d1/database/%s", rc.Identifier, databaseID)
+	_, err := api.makeRequestContext(ctx, http.MethodDelete, uri, nil)
+	if err != nil {
+		return fmt.Errorf("%s: %w", errMakeRequestError, err)
+	}
+	return nil
+}
+
+// GetD1Database returns a database for an account.
+//
+// API reference: https://developers.cloudflare.com/api/operations/cloudflare-d1-get-database
+func (api *API) GetD1Database(ctx context.Context, rc *ResourceContainer, databaseID string) (D1Database, error) {
+	if rc.Identifier == "" {
+		return D1Database{}, ErrMissingAccountID
+	}
+	uri := fmt.Sprintf("/accounts/%s/d1/database/%s", rc.Identifier, databaseID)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return D1Database{}, fmt.Errorf("%s: %w", errMakeRequestError, err)
+	}
+
+	var r D1DatabaseResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return D1Database{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return r.Result, nil
+}
+
+// QueryD1Database queries a database for an account.
+//
+// API reference: https://developers.cloudflare.com/api/operations/cloudflare-d1-query-database
+func (api *API) QueryD1Database(ctx context.Context, rc *ResourceContainer, params QueryD1Parameters) ([]D1Result, error) {
```

Comment:
the third parameter should be `<Method>Params`

```suggestion
func (api *API) QueryD1Database(ctx context.Context, rc *ResourceContainer, params QueryD1DatabaseParams) ([]D1Result, error) {
```